python/dgl/distributed/optim/pytorch/sparse_optim.py

```diff
@@ -35,16 +46,194 @@ def __init__(self, params, lr):
             self._rank = 0
             self._world_size = 1
 
+    def local_state_dict(self):
+        """Return the state pertaining to current rank of the optimizer.
+
+        Returns
+        -------
+        dict
+            Local state dict
+
+        See Also
+        --------
+        load_local_state_dict
+        """
+        lcoal_state_dict = {}
+        lcoal_state_dict["emb_states"] = {}
+        num_clients = dgl.distributed.get_num_client()
+        lcoal_state_dict["params"] = {"num_clients": num_clients}
```

Comment:
Why does `num_clients` go into the `params` entry?

Reply:
It is replaced by ```world_size``` now. And it is used to determine if there has a scale-in after reloading.